tests/periph_flash/main.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2014 Frank Holtz
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Flash interface test
+ *
+ * @author      Frank Holtz <frank-riot2015@holtznet.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "periph/flash.h"
+
+uint8_t print_test(char *text, uint32_t return_code, uint32_t error_code)
+{
+    if ((return_code == error_code) || (error_code == 65535)) {
+        printf("%s\tOK (%d)\n", text, return_code);
+        return 0;
+    }
+    else {
+        printf("%s\tError (%d)\n", text, return_code);
+        return 1;
+    }
+}
+
+int main(void)
+{
+    // Signature with space for alignment
+    char *testsignature = "RIOT-FLASH-TEST\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
+    int errors = 0, testsignaturelen = strlen(testsignature), counter = 0;
+    uint8_t *flashpage = flash_get_address(FLASH_NUM_PAGES - 1);
+
+    printf("\nFlash interface test...\nTest\t\t\t\t\tResult (RC)\n---------------------------------------------------\n");
+
+    errors = errors + print_test("Initializing flash interface\t", flash_init(), FLASH_ERROR_SUCCESS);
+    errors = errors + print_test("Compare signate with flash\t", memcmp(flashpage, testsignature,
+                                 strlen(testsignature)), 65535);
```

Comment:
That's correct. I have copied this line. Later in code i modify this variable for alignment.